modules/apps/object/object-api/src/main/java/com/liferay/object/exception/ObjectValidationRuleSettingValueException.java

```diff
@@ -12,8 +12,48 @@
  */
 public class ObjectValidationRuleSettingValueException extends PortalException {
 
-	public ObjectValidationRuleSettingValueException(String msg) {
-		super(msg);
+	public ObjectValidationRuleSettingValueException(String message) {
+		super(message);
+
+		_messageKey = message;
+	}
+
+	public ObjectValidationRuleSettingValueException(
+		String message, String messageKey) {
+
+		super(message);
+
+		_messageKey = messageKey;
+	}
+
+	public String getMessageKey() {
+		return _messageKey;
+	}
+
+	public static class CompositeKeyMustHaveFiveObjectFieldsAtMaximum
+		extends ObjectValidationRuleSettingValueException {
+
+		public CompositeKeyMustHaveFiveObjectFieldsAtMaximum() {
+			super(
+				"Add a maximum of five object fields to create unique " +
+					"composite keys",
+				"add-a-maximum-of-five-object-fields-to-create-unique-" +
+					"composite-keys");
+		}
+
+	}
+
+	public static class CompositeKeyMustHaveTwoObjectFieldsAtMinimum
```

Comment:
@mateussgn I prefer something more generic in case we change this min value in the future, something like `CompositeKeyMustHaveMinObjectFields`